src/board.py

```diff
@@ -138,18 +118,20 @@ def __getitem__(self, square: Square | Indices | None) -> Piece | None:
             The piece on the given square.
         """
         if square:
-            i, j = self._indices(square) if isinstance(square, Square) else square
-            return self._board[i][j]
+            if isinstance(square, str):
```

Comment:
No usage of `_square` here? If not, delete method.
Alternatively, do `square = Square(square)` even when `square` is a `Square`, it should work.

Reply:
Same thing, that shouldn't be there, I'll fix those things.